File: jiggles_native/app/src/main/java/com/example/cchiv/jiggles/model/Thread.java
package com.example.cchiv.jiggles.model;

import java.util.HashMap;
import java.util.List;

public class Thread extends FeedItem {

    private static final String TAG = "Thread";

    private String caption;

    public Thread(String _id, User author, String caption, String content, HashMap<String, Boolean>  likes, List<Reply> replies) {
        super(_id, author, content, likes, replies);

        this.caption = caption;
    }

    public String getCaption() {
        return caption;
    }

    @Override
    public int getViewType() {
        return ViewType.VIEW_THREAD;
    }
}

File: server/models/thread.js
const mongoose = require('mongoose');
const {ObjectId} = mongoose.Schema.Types;
const {ReplySchema} = require('./reply');

let ThreadSchema = new mongoose.Schema({
    author : {
        type : ObjectId,
        ref : 'User'
    },
    caption : {
        type :  String,
        default : ""
    },
    content : {
        type : String,
        require : true
    },
    likes : {
        type : Map,
        of : String,
        default : {}
    },
    replies: [ReplySchema]
});

let Thread = mongoose.model('Thread', ThreadSchema);

module.exports = {Thread, ThreadSchema};